src/components/Dashboard/Wrappers/gridstack/store.tsx

```diff
@@ -0,0 +1,50 @@
+import { useMantineTheme } from '@mantine/core';
+import create from 'zustand';
+import { useConfigContext } from '../../../../config/provider';
+
+export const useGridstackStore = create<GridstackStoreType>((set, get) => ({
+  mainAreaWidth: null,
+  currentShapeSize: null,
+  setMainAreaWidth: (w: number) =>
+    set((v) => ({ ...v, mainAreaWidth: w, currentShapeSize: getCurrentShapeSize(w) })),
+}));
+
+interface GridstackStoreType {
+  mainAreaWidth: null | number;
+  currentShapeSize: null | 'sm' | 'md' | 'lg';
+  setMainAreaWidth: (width: number) => void;
+}
+
+export const useNamedWrapperColumnCount = (): 'small' | 'medium' | 'large' | null => {
+  const mainAreaWidth = useGridstackStore((x) => x.mainAreaWidth);
+  const { sm, xl } = useMantineTheme().breakpoints;
+  if (!mainAreaWidth) return null;
+
+  if (mainAreaWidth >= xl) return 'large';
+
+  if (mainAreaWidth >= sm) return 'medium';
+
+  return 'small';
+};
+
+export const useWrapperColumnCount = () => {
+  const { config } = useConfigContext();
+  const numberOfSidebars =
+    (config?.settings.customization.layout.enabledLeftSidebar ? 1 : 0) +
+    (config?.settings.customization.layout.enabledRightSidebar ? 1 : 0);
+
+  switch (useNamedWrapperColumnCount()) {
```

Comment:
This breaks everything and should be reverted to the previous one

Reply:
I confirm this. It's a stupid solution and could break many things

Reply:
okok we can revert it but let's increase the base amount of columns to 8 or 9, what do you guys think?

Reply:
Yes, sounds like a good idea